pinot-spi/src/main/java/org/apache/pinot/spi/config/table/TableConfig.java

```diff
@@ -101,6 +102,9 @@ public class TableConfig extends BaseJsonConfig {
   @JsonPropertyDescription(value = "Configs for Table config tuner")
   private List<TunerConfig> _tunerConfigList;
 
+  @JsonPropertyDescription(value = "Name of the table-group")
+  private String _tableGroupName;
```

Comment:
One table group can refer to one `InstanceAssignment`, but one table might be associated with multiple `InstanceAssignment`. Should we keep a map from `InstancePartitionsType` to group name, or consider adding group name as a field of `InstanceAssignmentConfig`?

Reply:
Based on our conversation, we went with creating a map in table-config called "instancePartitionsMap" which is from instance partitions type to an InstancePartitions name (could be a group or an existing table)